### PR TITLE
Add doc strings for all Mixin classes and functions

### PR DESCRIPTION
**Pull request by jashparekh**
## Description

- Add doc comments for all the functions in Mixin classes

### CODE REVIEW

File: instagrapi/mixins/collection.py [hunk 1]
```diff
@@ -28,16 +40,37 @@ def collections(self) -> list:
...
+        Get media in a collection by collection pk
+
+        Parameters
+        ----------
+        collection_pk : int
+            Unique identifier of a Collection
+        amount : int, optional
+            Maximum number of media to return, default is 21
+        last_media_pk : int, optional
+            Last PK user has seen, function will return medias after this pk. Default is 0
```
Comment (jashparekh) [inline instagrapi/mixins/collection.py:68]:
```suggestion
        collection_pk: int
            Unique identifier of a Collection
        amount: int, optional
            Maximum number of media to return, default is 21
        last_media_pk: int, optional
            Last PK user has seen, function will return medias after this pk. Default is 0
```

File: instagrapi/mixins/comment.py [hunk 2]
```diff
@@ -11,9 +11,23 @@
...
+    """
 
     def media_comments(self, media_id: str) -> List[Comment]:
-        """Get list of comments for media
+        """
+        Get comments on a media
+
+        Parameters
+        ----------
+        media_id : str
```
Comment (jashparekh) [inline instagrapi/mixins/comment.py:24]:
```suggestion
        media_id: str
```

File: instagrapi/mixins/comment.py [hunk 3]
```diff
@@ -38,7 +52,20 @@ def media_comments(self, media_id: str) -> List[Comment]:
         return comments
 
     def media_comment(self, media_id: str, text: str) -> Comment:
-        """Comment media
+        """
+        Post a comment on a media
+
+        Parameters
+        ----------
+        media_id : str
```
Comment (jashparekh) [inline instagrapi/mixins/comment.py:60]:
```suggestion
        media_id: str
```

File: instagrapi/mixins/comment.py [hunk 4]
```diff
@@ -58,7 +85,20 @@ def media_comment(self, media_id: str, text: str) -> Comment:
         return extract_comment(result["comment"])
 
     def comment_like(self, comment_pk: int, revert: bool = False) -> bool:
-        """Like comment
+        """
+        Like a comment on a media
+
+        Parameters
+        ----------
+        comment_pk : str
```
Comment (jashparekh) [inline instagrapi/mixins/comment.py:93]:
```suggestion
        comment_pk: str
```

File: instagrapi/mixins/comment.py [hunk 5]
```diff
@@ -75,6 +115,17 @@ def comment_like(self, comment_pk: int, revert: bool = False) -> bool:
         return result['status'] == 'ok'
 
     def comment_unlike(self, comment_pk: str) -> bool:
-        """Unlike comment
+        """
+        Unlike a comment on a media
+
+        Parameters
+        ----------
+        comment_pk : str
```
Comment (jashparekh) [inline instagrapi/mixins/comment.py:123]:
```suggestion
        comment_pk: str
```

File: instagrapi/mixins/direct.py [hunk 6]
```diff
@@ -8,9 +8,23 @@
...
+    """
 
     def direct_threads(self, amount: int = 20) -> List[DirectThread]:
-        """Return last threads
+        """
+        Get direct message threads
+
+        Parameters
+        ----------
+        amount : int, optional
```
Comment (jashparekh) [inline instagrapi/mixins/direct.py:21]:
```suggestion
        amount: int, optional
```

File: instagrapi/mixins/direct.py [hunk 7]
```diff
@@ -37,7 +51,21 @@ def direct_threads(self, amount: int = 20) -> List[DirectThread]:
...
-        """Return full information by thread
+        """
+        Get all the information about a Direct Message thread
+
+        Parameters
+        ----------
+        thread_id: int
+            Unique identifier of a Direct Message thread
+
+        amount : int, optional
```
Comment (jashparekh) [inline instagrapi/mixins/direct.py:62]:
```suggestion
        amount: int, optional
```

File: instagrapi/mixins/direct.py [hunk 8]
```diff
@@ -67,19 +95,64 @@ def direct_thread(self, thread_id: int, amount: int = 20) -> DirectThread:
...
-        """Fetch list of messages by thread (helper)
+        """
+        Get all the messages from a thread
+
+        Parameters
+        ----------
+        thread_id: int
+            Unique identifier of a Direct Message thread
+
+        amount : int, optional
```
Comment (jashparekh) [inline instagrapi/mixins/direct.py:106]:
```suggestion
        amount: int, optional
```

File: instagrapi/mixins/direct.py [hunk 9]
```diff
@@ -67,19 +95,64 @@ def direct_thread(self, thread_id: int, amount: int = 20) -> DirectThread:
...
-        """Send message
+        """
+        Post a message on a Direct Message thread
+
+        Parameters
+        ----------
+        thread_id: int
+            Unique identifier of a Direct Message thread
+
+        text : str
```
Comment (jashparekh) [inline instagrapi/mixins/direct.py:126]:
```suggestion
        text: str
```

File: instagrapi/mixins/direct.py [hunk 10]
```diff
@@ -67,19 +95,64 @@ def direct_thread(self, thread_id: int, amount: int = 20) -> DirectThread:
...
         return self.direct_send(text, [], [int(thread_id)])
 
     def direct_send(self, text: str, user_ids: List[int] = [], thread_ids: List[int] = []) -> DirectMessage:
-        """Send message
+        """
+        Send a direct message to list of users or threads
+
+        Parameters
+        ----------
+        text : str
```
Comment (jashparekh) [inline instagrapi/mixins/direct.py:143]:
```suggestion
        text: str
```

File: instagrapi/mixins/hashtag.py [hunk 11]
```diff
@@ -12,9 +12,26 @@
...
+    """
 
     def hashtag_info_a1(self, name: str, max_id: str = None) -> Hashtag:
-        """Get info (id, name, profile_pic_url)
+        """
+        Get information about a hashtag
+
+        Parameters
+        ----------
+        name : str
```
Comment (jashparekh) [inline instagrapi/mixins/hashtag.py:25]:
```suggestion
        name: str
```

File: instagrapi/mixins/hashtag.py [hunk 12]
```diff
@@ -23,7 +40,24 @@ def hashtag_info_a1(self, name: str, max_id: str = None) -> Hashtag:
...
+            Name of the hashtag
+
+        amount : int, optional
+            Maximum number of media to return, default is 12
```
Comment (jashparekh) [inline instagrapi/mixins/hashtag.py:52]:
```suggestion
        name: str
            Name of the hashtag

        amount: int, optional
            Maximum number of media to return, default is 12
```

File: instagrapi/mixins/hashtag.py [hunk 13]
```diff
@@ -38,13 +72,35 @@ def hashtag_info_gql(self, name: str, amount: int = 12, end_cursor: str = None)
         return extract_hashtag_gql(data["hashtag"])
 
     def hashtag_info_v1(self, name: str) -> Hashtag:
-        """Get info (id, name, profile_pic_url)
+        """
+        Get information about a hashtag
+
+        Parameters
+        ----------
+        name : str
```
Comment (jashparekh) [inline instagrapi/mixins/hashtag.py:80]:
```suggestion
        name: str
```

File: instagrapi/mixins/hashtag.py [hunk 14]
```diff
@@ -38,13 +72,35 @@ def hashtag_info_gql(self, name: str, amount: int = 12, end_cursor: str = None)
...
         return extract_hashtag_v1(result)
 
     def hashtag_info(self, name: str) -> Hashtag:
-        """Get info (id, name, profile_pic_url)
+        """
+        Get information about a hashtag
+
+        Parameters
+        ----------
+        name : str
```
Comment (jashparekh) [inline instagrapi/mixins/hashtag.py:97]:
```suggestion
        name: str
```

File: instagrapi/mixins/hashtag.py [hunk 15]
```diff
@@ -55,7 +111,18 @@ def hashtag_info(self, name: str) -> Hashtag:
         return hashtag
 
     def hashtag_related_hashtags(self, name: str) -> List[Hashtag]:
-        """Get related hashtags
+        """
+        Get related hashtags from a hashtag
+
+        Parameters
+        ----------
+        name : str
```
Comment (jashparekh) [inline instagrapi/mixins/hashtag.py:119]:
```suggestion
        name: str
```

File: instagrapi/mixins/hashtag.py [hunk 16]
```diff
@@ -64,9 +131,24 @@ def hashtag_related_hashtags(self, name: str) -> List[Hashtag]:
...
+        Get medias for a hashtag
+
+        Parameters
+        ----------
+        name : str
+            Name of the hashtag
+        amount : int, optional
+            Maximum number of media to return, default is 27
+        tab_key : str, optional
+            Tab Key, default value is ""
```
Comment (jashparekh) [inline instagrapi/mixins/hashtag.py:144]:
```suggestion
        name: str
            Name of the hashtag
        amount: int, optional
            Maximum number of media to return, default is 27
        tab_key: str, optional
            Tab Key, default value is ""
```

File: instagrapi/mixins/hashtag.py [hunk 17]
```diff
@@ -100,7 +182,24 @@ def hashtag_medias_a1(self, name: str, amount: int = 27, tab_key: str = '') -> L
...
+        Parameters
+        ----------
+        name : str
+            Name of the hashtag
+
+        amount : int, optional
+            Maximum number of media to return, default is 27
+
+        tab_key : str, optional
+            Tab Key, default value is ""
```
Comment (jashparekh) [inline instagrapi/mixins/hashtag.py:197]:
```suggestion
        name: str
            Name of the hashtag
        amount: int, optional
            Maximum number of media to return, default is 27
        tab_key: str, optional
            Tab Key, default value is ""
```

File: instagrapi/mixins/hashtag.py [hunk 18]
```diff
@@ -138,20 +237,62 @@ def hashtag_medias_v1(self, name: str, amount: int = 27, tab_key: str = '') -> L
...
+        """
+        Get top medias for a hashtag
+
+        Parameters
+        ----------
+        name : str
+            Name of the hashtag
+
+        amount : int, optional
+            Maximum number of media to return, default is 9
```
Comment (jashparekh) [inline instagrapi/mixins/hashtag.py:249]:
```suggestion
        name: str
            Name of the hashtag
        amount: int, optional
            Maximum number of media to return, default is 9
```

File: instagrapi/mixins/hashtag.py [hunk 19]
```diff
@@ -138,20 +237,62 @@ def hashtag_medias_v1(self, name: str, amount: int = 27, tab_key: str = '') -> L
...
```
Comment (jashparekh) [inline instagrapi/mixins/hashtag.py:271]:
```suggestion
        name: str
            Name of the hashtag
        amount: int, optional
            Maximum number of media to return, default is 9
```

File: instagrapi/mixins/hashtag.py [hunk 20]
```diff
@@ -138,20 +237,62 @@ def hashtag_medias_v1(self, name: str, amount: int = 27, tab_key: str = '') -> L
...
```
Comment (jashparekh) [inline instagrapi/mixins/hashtag.py:290]:
```suggestion
        name: str
            Name of the hashtag
        amount: int, optional
            Maximum number of media to return, default is 9
```

File: instagrapi/mixins/hashtag.py [hunk 21]
```diff
@@ -167,20 +308,62 @@ def hashtag_medias_top(self, name: str, amount: int = 9) -> List[Media]:
...
+        """
+        Get recent medias for a hashtag
+
+        Parameters
+        ----------
+        name : str
+            Name of the hashtag
+
+        amount : int, optional
+            Maximum number of media to return, default is 71
```
Comment (jashparekh) [inline instagrapi/mixins/hashtag.py:320]:
```suggestion
        name: str
            Name of the hashtag
        amount: int, optional
            Maximum number of media to return, default is 71
```

File: instagrapi/mixins/hashtag.py [hunk 22]
```diff
@@ -167,20 +308,62 @@ def hashtag_medias_top(self, name: str, amount: int = 9) -> List[Media]:
...
```
Comment (jashparekh) [inline instagrapi/mixins/hashtag.py:342]:
```suggestion
        name: str
            Name of the hashtag
        amount: int, optional
            Maximum number of media to return, default is 71
```

File: instagrapi/mixins/hashtag.py [hunk 23]
```diff
@@ -167,20 +308,62 @@ def hashtag_medias_top(self, name: str, amount: int = 9) -> List[Media]:
...
```
Comment (jashparekh) [inline instagrapi/mixins/hashtag.py:361]:
```suggestion
        name: str
            Name of the hashtag
        amount: int, optional
            Maximum number of media to return, default is 71
```

File: instagrapi/mixins/igtv.py [hunk 24]
```diff
@@ -18,15 +18,51 @@
...
+    Helpers to download IGTV videos
+    """
 
     def igtv_download(self, media_pk: int, folder: Path = "") -> str:
+        """
+        Download IGTV video
+
+        Parameters
+        ----------
+        media_pk : int
```
Comment (jashparekh) [inline instagrapi/mixins/igtv.py:31]:
```suggestion
        media_pk: int
```

File: instagrapi/mixins/igtv.py [hunk 25]
```diff
@@ -18,15 +18,51 @@
...
+        """
         return self.video_download(media_pk, folder)
 
     def igtv_download_by_url(self, url: str, filename: str = "", folder: Path = "") -> str:
+        """
+        Download IGTV video using URL
+
+        Parameters
+        ----------
+        url : str
```
Comment (jashparekh) [inline instagrapi/mixins/igtv.py:49]:
```suggestion
        url: str
```

File: instagrapi/mixins/igtv.py [hunk 26]
```diff
@@ -38,18 +74,30 @@ def igtv_upload(
...
+        Parameters
+        ----------
+        path : Path
+            Path to IGTV file
+        title : str
+            Title of the video
+        caption : str
+            Media caption
+        thumbnail : Path, optional
+            Path to thumbnail for IGTV. Default value is None, and it generates a thumbnail
```
Comment (jashparekh) [inline instagrapi/mixins/igtv.py:89]:
```suggestion
        path: Path
            Path to IGTV file
        title: str
            Title of the video
        caption: str
            Media caption
        thumbnail: Path, optional
            Path to thumbnail for IGTV. Default value is None, and it generates a thumbnail
```

File: instagrapi/mixins/igtv.py [hunk 27]
```diff
@@ -144,17 +192,35 @@ def igtv_configure(
...
-        :param location:   Location
+    ) -> Dict:
+        """
+        Post Configure IGTV (send caption, thumbnail and more to Instagram)
+
+        Parameters
+        ----------
+        upload_id: str
+            Unique identifier for a IGTV video
+        thumbnail : Path
```
Comment (jashparekh) [inline instagrapi/mixins/igtv.py:203]:
```suggestion
        thumbnail: Path
```

File: instagrapi/mixins/igtv.py [hunk 28]
```diff
@@ -144,17 +192,35 @@ def igtv_configure(
...
+        width: int
+            Width of the video in pixels
+        height: int
+            Height of the video in pixels
+        duration: int
+            Duration of the video in seconds
+        title : str
+            Title of the video
+        caption : str
+            Media caption
```
Comment (jashparekh) [inline instagrapi/mixins/igtv.py:214]:
```suggestion
        title: str
            Title of the video
        caption: str
            Media caption
```

File: instagrapi/mixins/igtv.py [hunk 29]
```diff
@@ -190,9 +256,21 @@ def igtv_configure(
...
-    """Analyze and crop thumbnail if need
     """
+    Analyze and crop thumbnail if need
 
+    Parameters
+    ----------
+    path : Path
+        Path to the video
+    thumbnail : Path
+        Path to thumbnail for IGTV
```
Comment (jashparekh) [inline instagrapi/mixins/igtv.py:267]:
```suggestion
    path: Path
        Path to the video
    thumbnail: Path
        Path to thumbnail for IGTV
```

File: instagrapi/mixins/igtv.py [hunk 30]
```diff
@@ -210,7 +288,18 @@ def analyze_video(path: Path, thumbnail: Path = None) -> tuple:
 
 
 def crop_thumbnail(path: Path) -> bool:
-    """Crop IGTV thumbnail with save height
+    """
+    Analyze and crop thumbnail if need
+
+    Parameters
+    ----------
+    path : Path
```
Comment (jashparekh) [inline instagrapi/mixins/igtv.py:296]:
```suggestion
    path: Path
```

File: instagrapi/mixins/insights.py [hunk 31]
```diff
@@ -13,15 +17,27 @@ def insights_media_feed_all(
...
-        :rtype: list
+    ) -> List[Dict]:
+        """
+        Get insights for all medias from feed with page iteration with cursor and sleep timeout
+
+        Parameters
+        ----------
+        post_type: str, optional
+            Types of posts, default is "ALL"
+        time_frame : str, optional
```
Comment (jashparekh) [inline instagrapi/mixins/insights.py:28]:
```suggestion
        time_frame: str, optional
```

File: instagrapi/mixins/insights.py [hunk 32]
```diff
@@ -13,15 +17,27 @@ def insights_media_feed_all(
...
-        :return: List with media insights
-        :rtype: list
+    ) -> List[Dict]:
+        """
+        Get insights for all medias from feed with page iteration with cursor and sleep timeout
+
+        Parameters
+        ----------
+        post_type: str, optional
+            Types of posts, default is "ALL"
```
Comment (jashparekh) [inline instagrapi/mixins/insights.py:27]:
```suggestion
            Types of posts, default is "ALL"
            Options: ("ALL", "CAROUSEL_V2", "IMAGE", "SHOPPING", "VIDEO")
```

File: instagrapi/mixins/insights.py [hunk 33]
```diff
@@ -13,15 +17,27 @@ def insights_media_feed_all(
...
+        time_frame : str, optional
+            Time frame to pull media insights, default is "TWO_YEARS"
```
Comment (jashparekh) [inline instagrapi/mixins/insights.py:29]:
```suggestion
            Time frame to pull media insights, default is "TWO_YEARS"
            Options: ("ONE_WEEK", "ONE_MONTH", "THREE_MONTHS", "SIX_MONTHS", "ONE_YEAR", "TWO_YEARS")
```

File: instagrapi/mixins/insights.py [hunk 34]
```diff
@@ -13,15 +17,27 @@ def insights_media_feed_all(
...
+        data_ordering: str, optional
+            Ordering strategy for the data, default is "REACH_COUNT"
```
Comment (jashparekh) [inline instagrapi/mixins/insights.py:31]:
```suggestion
            Ordering strategy for the data, default is "REACH_COUNT"
            Options: ("REACH_COUNT", "LIKE_COUNT", "FOLLOW", "SHARE_COUNT", "BIO_LINK_CLICK", "COMMENT_COUNT", "IMPRESSION_COUNT", "PROFILE_VIEW", "VIDEO_VIEW_COUNT", "SAVE_COUNT"...)
```

File: instagrapi/mixins/insights.py [hunk 35]
```diff
@@ -121,11 +145,19 @@ def insights_account(self) -> dict:
...
-        :param media_pk:  Media id
-        :return: Dict with insights data
-        :rtype: dict
+    def insights_media(self, media_pk: int) -> Dict:
+        """
+        Get insights data for media
+
+        Parameters
+        ----------
+        media_pk : int
```
Comment (jashparekh) [inline instagrapi/mixins/insights.py:154]:
```suggestion
        media_pk: int
```

File: instagrapi/mixins/media.py [hunk 36]
```diff
@@ -152,8 +272,27 @@ def media_edit(
...
+        """
+        Edit caption for media
+
+        Parameters
+        ----------
+        media_id: str
+            Unique identifier of the media
+        caption: str
+            Media caption
+        title : str
```
Comment (jashparekh) [inline instagrapi/mixins/media.py:285]:
```suggestion
        title: str
```

File: instagrapi/mixins/media.py [hunk 37]
```diff
@@ -188,19 +327,54 @@ def media_edit(
...
         )
 
     def media_like(self, media_id: str, revert: bool = False) -> bool:
-        """Like media
+        """
+        Like a media
+
+        Parameters
+        ----------
+        media_id : str
```
Comment (jashparekh) [inline instagrapi/mixins/media.py:369]:
```suggestion
        media_id: str
```

File: instagrapi/mixins/media.py [hunk 38]
```diff
@@ -220,6 +394,17 @@ def media_like(self, media_id: str, revert: bool = False) -> bool:
         return result['status'] == 'ok'
 
     def media_unlike(self, media_id: str) -> bool:
-        """Unlike media
+        """
+        Unlike a media
+
+        Parameters
+        ----------
+        media_id : str
```
Comment (jashparekh) [inline instagrapi/mixins/media.py:402]:
```suggestion
        media_id: str
```

File: instagrapi/mixins/user.py [hunk 39]
```diff
@@ -151,8 +278,21 @@ def user_info(self, user_id: int, use_cache: bool = True) -> User:
...
-        """Return list of following users (without authorization)
+    def user_following_gql(self, user_id: int, amount: int = 0) -> List[User]:
+        """
+        Get user's followers information
+
+        Parameters
+        ----------
+        user_id: int
+            User id of an instagram account
+        amount : int, optional
```
Comment (jashparekh) [inline instagrapi/mixins/user.py:289]:
```suggestion
        amount: int, optional
```

File: instagrapi/mixins/user.py [hunk 40]
```diff
@@ -189,8 +329,21 @@ def user_following_gql(self, user_id: int, amount: int = 0) -> list:
...
-        """Return list of following users (with authorization)
+    def user_following_v1(self, user_id: int, amount: int = 0) -> List[User]:
+        """
+        Get user's followers information
+
+        Parameters
+        ----------
+        user_id: int
+            User id of an instagram account
+        amount : int, optional
```
Comment (jashparekh) [inline instagrapi/mixins/user.py:340]:
```suggestion
        amount: int, optional
```

File: instagrapi/mixins/user.py [hunk 41]
```diff
@@ -214,7 +367,22 @@ def user_following_v1(self, user_id: int, amount: int = 0) -> list:
...
+        """
+        Get user's followers information
+
+        Parameters
+        ----------
+        user_id: int
+            User id of an instagram account
+        use_cache: bool, optional
+            Whether or not to use information from cache, default value is True
+        amount : int, optional
```
Comment (jashparekh) [inline instagrapi/mixins/user.py:379]:
```suggestion
        amount: int, optional
```

File: instagrapi/mixins/user.py [hunk 42]
```diff
@@ -231,8 +399,21 @@ def user_following(self, user_id: int, use_cache: bool = True, amount: int = 0)
...
-        """Return list of followers users (with auth)
+    def user_followers_v1(self, user_id: int, amount: int = 0) -> List[User]:
+        """
+        Get user's followers information
+
+        Parameters
+        ----------
+        user_id: int
+            User id of an instagram account
+        amount : int, optional
```
Comment (jashparekh) [inline instagrapi/mixins/user.py:410]:
```suggestion
        amount: int, optional
```

File: instagrapi/mixins/user.py [hunk 43]
```diff
@@ -250,7 +431,22 @@ def user_followers_v1(self, user_id: int, amount: int = 0) -> list:
...
+        """
+        Get user's followers
+
+        Parameters
+        ----------
+        user_id: int
+            User id of an instagram account
+        use_cache: bool, optional
+            Whether or not to use information from cache, default value is True
+        amount : int, optional
```
Comment (jashparekh) [inline instagrapi/mixins/user.py:443]:
```suggestion
        amount: int, optional
```

File: instagrapi/mixins/user.py [hunk 44]
```diff
@@ -287,13 +503,20 @@ def user_unfollow(self, user_id: int) -> bool:
...
-        :param user_id: Profile user id in instagram
-        :param amount: Count of medias for fetching (by default instagram return 50)
-        :param sleep: Timeout between requests
-        :return: List of medias for profile
+        Get a user's media
+
+        Parameters
+        ----------
+        user_id: int
+        amount : int, optional
```
Comment (jashparekh) [inline instagrapi/mixins/user.py:511]:
```suggestion
        amount: int, optional
```

File: instagrapi/mixins/user.py [hunk 45]
```diff
@@ -326,9 +549,19 @@ def user_medias_gql(self, user_id: int, amount: int = 50, sleep: int = 2) -> Lis
...
-        """Get all medias by user_id via Private API
-        :user_id: User ID
-        :amount: By default instagram return 18 items by each request
+        """
+        Get a user's media
+
+        Parameters
+        ----------
+        user_id: int
+        amount : int, optional
```
Comment (jashparekh) [inline instagrapi/mixins/user.py:558]:
```suggestion
        amount: int, optional
```

File: instagrapi/mixins/user.py [hunk 46]
```diff
@@ -358,8 +591,19 @@ def user_medias_v1(self, user_id: int, amount: int = 18) -> List[Media]:
...
     def user_medias(self, user_id: int, amount: int = 50) -> List[Media]:
-        """Get all medias by user_id
-        First, through the Public API, then through the Private API
+        """
+        Get a user's media
+
+        Parameters
+        ----------
+        user_id: int
+        amount : int, optional
```
Comment (jashparekh) [inline instagrapi/mixins/user.py:600]:
```suggestion
        amount: int, optional
```

**Comment by adw0rd** [comment]
Big thanks!

**Pull request merged**